### PR TITLE
add labels method for subclasses to override

### PR DESCRIPTION
**Pull request by ihurrahi**
No behavior change. Exposes a method for subclasses of the Middleware to provide custom labels based on the `env`

### CODE REVIEW

File: lib/prometheus_exporter/middleware.rb [hunk 1]
```diff
@@ -44,14 +44,24 @@ def call(env)
...
+    labels = custom_labels(env)
+    if labels.present? && labels.length > 0
+      obj = obj.merge(custom_labels: labels)
+    end
+    
+    @client.send_json(obj)
+  end
+
+  def custom_labels(env)
+    {}
```
Comment (walkerdb) [inline lib/prometheus_exporter/middleware.rb:65]: should this do something other than return an empty object?

Comment (ihurrahi) [reply lib/prometheus_exporter/middleware.rb:65]: what do you mean? this shouldn't change any current behavior in the library besides make it easy for subclasses to provide custom_labels

Comment (walkerdb) [reply lib/prometheus_exporter/middleware.rb:65]: ahh gotcha. Sorry, confused by languages without explicit interfaces..

Comment (walkerdb) [reply lib/prometheus_exporter/middleware.rb:65]: Might be worth a comment or something detailing that this method exists largely to be overridden by its subclass

**Pull request merged**